text/0000-uninstall-node.md

```diff
@@ -0,0 +1,64 @@
+- Feature Name: uninstall_node
+- Start Date: 2024-08-21
+- RFC PR:[https://github.com/volta-cli/volta/pull/1882]
+- Volta Issue:
+
+# Summary
+
+[summary]: #summary
+
+Add support for uninstalling node to `volta uninstall`
+
+# Motivation
+
+[motivation]: #motivation
+
+Currently, Volta doesn't support uninstalling node due to this main reasons(as discussed in [#327](!https://github.com/volta-cli/volta/issues/327)): Its main purpose is to save disk space, nowadays disk size is not a big problem in many cases. So this task is in a low priority.But community has this demand since 2019, they need to manually remove some files to achieve that.Add support for uninstalling node is more convinent in this case, btw since we have `install` here we should also have `uninstall` here.
+
+# Pedagogy
+
+[pedagogy]: #pedagogy
+
+The user's intuition and way of treating it are opposite to `install`. Since `install` fetch node image, unpack it and set it as default, `unintall` should remove those things.
+
+# Details
+
+[details]: #details
+
+The new command will have the structure:
+
+```
+volta uninstall node@version
+```
+
+totally the same with `install`.
+
+Some examples:
+
+```
+volta uninstall node@20.16.0
+```
+
+```
+volta uninstall node@lts
+```
+
+By add a `uninstall` to `Tool` trait and move uninstall logic from `Spec` to `Node, Package, Npm, Yarn, Pnpm`, it would resolve version with `node::resolve` first, and then remove tar.gz file and npm file in `home.node_inventory_dir()` and node in `home.node_image_root_dir()`
+
+# Critique
+
+[critique]: #critique
+
+In fact we want `uninstall` to save disk space or just to remove those things we don't need , it seems like a command like `prune` or `clean` is more suitable, which tracks every tool's status, and when `prune` is runned, we can clean all those tools that is no longer used.
```

Comment:
This is a good call out! I have a separate RFC up (which I'm going to be updating soon, hopefully today) to deprecate the `install` command and instead use a different verb: https://github.com/volta-cli/rfcs/pull/52

In that context, would it be confusing to have a `clean` or `prune` command that works similarly to the proposed `uninstall` but uses a different verb? I.e. you would still need to provide a tool name (and possibly version, depending on the above discussion), but it would be `volta clean node@10` instead of `volta uninstall node@10`

Reply:
In that case, since ```install``` is deprecated, there is no need for a ```uninstall``` verb. How about rename ```uninstall``` to ```remove``` and support ```volta remove node@version``` to manually remove node image, no more other verbs?